utils/utils.c

```diff
@@ -392,3 +392,81 @@ char * strjoin(char *left, char *right, char *delim)
 	strcpy(new + len - rlen - 1, right);
 	return new;
 }
+
+#define QUOTE '\''
+#define DQUOTE '"'
+
+static int setargs(char *args, char **argv)
+{
+	int count = 0;
+
+	/* ignore start spaces */
+	while (isspace(*args))
+		++args;
+
+	while (*args) {
+		/* consider quotes and update argv */
+		if (*args == QUOTE) {
+			if (argv)
+				argv[count] = args + 1;
+			while (*args != QUOTE)
+				++args;
+		} else if (*args == DQUOTE) {
+			if (argv)
+				argv[count] = args + 1;
+			while (*args != DQUOTE)
+				++args;
+		} else if (argv) {
+			argv[count] = args;
+		}
+		/* read characters until '\0' or space */
+		while (*args && !isspace(*args))
+			++args;
+		/* set '\0' rather than space */
+		if (argv && *args)
+			*args++ = '\0';
+		/* ignore margin spaces */
+		while (isspace(*args))
+			++args;
+		/* count up argument */
+		count++;
+	}
+
+	if (argv)
+		argv[count] = NULL;
+	return count;
+}
+
+#undef QUOTE
+#undef DQUOTE
+
+char **parse_cmdline(char *cmd, int *argc)
+{
+	char **argv = NULL;
+	int argn = 0;
+
+	if (cmd && *cmd
```

Comment:
done.

Also, fixed what I implemented wrong way. (I tested this on separated one but way of test was wrong before)